### PR TITLE
[Ingest Manager] Send datastreams fields

### PR DESCRIPTION
**Pull request by michalpristas**
## What does this PR do?

Adds `datastream.*` fields as for request described here https://github.com/elastic/integrations/pull/213 and https://github.com/elastic/elasticsearch/pull/60592

## Why is it important?

Desciben in an issue here https://github.com/elastic/integrations/pull/213 and https://github.com/elastic/elasticsearch/pull/60592

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

cc @ruflin

### CODE REVIEW

File: x-pack/elastic-agent/pkg/agent/operation/monitoring.go [hunk 1]
```diff
@@ -232,6 +242,16 @@ func (o *Operator) getMonitoringFilebeatConfig(output interface{}) (map[string]i
 							},
 						},
 					},
+					{
+						"add_fields": map[string]interface{}{
+							"target": "datastream",
+							"fields": map[string]interface{}{
+								"type":      "logs",
+								"dataset":   "elastic.agent",
```
Comment (blakerouse) [inline x-pack/elastic-agent/pkg/agent/operation/monitoring.go:250]: This is wrong, should be `fmt.Sprintf("elastic.agent.%s", name)`

File: x-pack/elastic-agent/pkg/agent/operation/monitoring.go [hunk 2]
```diff
@@ -232,6 +242,16 @@ func (o *Operator) getMonitoringFilebeatConfig(output interface{}) (map[string]i
...
+								"namespace": "default",
+							},
+						},
+					},
 					{
 						"add_fields": map[string]interface{}{
```
Comment (blakerouse) [inline x-pack/elastic-agent/pkg/agent/operation/monitoring.go:256]: You missed adding it to `getMonitoringMetricbeatConfig`

File: x-pack/elastic-agent/pkg/agent/operation/monitoring.go (hunk 1 above)
Comment (michalpristas) [reply x-pack/elastic-agent/pkg/agent/operation/monitoring.go]: @blakerouse updated

File: x-pack/elastic-agent/pkg/agent/operation/monitoring.go (hunk 2 above)
Comment (michalpristas) [reply x-pack/elastic-agent/pkg/agent/operation/monitoring.go:256]: thanks, i totally missed that

**Comment by ruflin** [comment]
@michalpristas did not test it locally but remember sometimes the processors for the monitoring data went missing. Just double checking.

**Comment by ruflin** [comment]
@michalpristas Just saw that this PR is against master. Most importantly we will need the change in 7.9. 7.10 will be only datastreams so not a direct backport. Perhaps better to point it to 7.9 directly?

**Pull request merged**